src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -160,6 +161,18 @@ public class Session {
             HostToClientWorker(_process.StandardOutput.BaseStream, hostEnd).DoNotWait();
         }
 
+        private void StartSession() {
+            _process.Start();
+            _process.WaitForExit(250);
+            if (_process.HasExited && _process.ExitCode != 0) {
+                var message = ErrorCodeConverter.MessageFromErrorCode(_process.ExitCode);
```

Comment:
I know. However, OS failures come in the same code. Ex, missing dll comes as C0000135 (NTSTATUS == module missing). I think user terminating host with odd error codes is less common.  We could theoretically use CreateProcess and separate exit code from actual last error, but that is quite a bit more work...